src/test/java/com/example/onlinebookstore/controller/BookControllerTest.java

```diff
@@ -0,0 +1,284 @@
+    private static BookDto expectedBookDto2;
+    @Autowired
+    private ObjectMapper objectMapper;
+    @Autowired
+    private BookRepository bookRepository;
+
+    @SneakyThrows
+    @BeforeAll
+    static void beforeAll(@Autowired WebApplicationContext webApplicationContext,
+                          @Autowired DataSource dataSource) {
+        mockMvc = MockMvcBuilders.webAppContextSetup(webApplicationContext)
+                .apply(springSecurity())
+                .build();
+        teardown(dataSource);
+        try (Connection connection = dataSource.getConnection()) {
+            ScriptUtils.executeSqlScript(
+                    connection,
+                    new ClassPathResource("db/book-category-insert.sql")
+            );
+        }
+
+        expectedBookDto = new BookDto();
+        expectedBookDto.setId(1L);
+        expectedBookDto.setTitle("Book 1");
+        expectedBookDto.setAuthor("Author 1");
+        expectedBookDto.setIsbn("ISBN-001");
+        expectedBookDto.setPrice(BigDecimal.valueOf(20));
+        expectedBookDto.setDescription("Description for Book 1");
+        expectedBookDto.setCoverImage("image1.jpg");
+        expectedBookDto.setCategoriesIds(Set.of(1L));
+
+        expectedBookDto2 = new BookDto();
+        expectedBookDto2.setId(2L);
+        expectedBookDto2.setTitle("Book 2");
+        expectedBookDto2.setAuthor("Author 2");
+        expectedBookDto2.setIsbn("ISBN-002");
+        expectedBookDto2.setPrice(BigDecimal.valueOf(30));
+        expectedBookDto2.setDescription("Description for Book 2");
+        expectedBookDto2.setCoverImage("image2.jpg");
+        expectedBookDto2.setCategoriesIds(Set.of(2L));
+
+        createBookRequest = new CreateBookRequestDto();
+        createBookRequest.setTitle("Pride and Prejudice");
+        createBookRequest.setAuthor("Jane Austen");
+        createBookRequest.setIsbn("978-1-60309-515-0");
+        createBookRequest.setPrice(new BigDecimal(13));
+        createBookRequest.setDescription("A classic romance novel");
+        createBookRequest.setCoverImage("https://pride.com/prejudice.jpg");
+        Set<Long> categoryIdsForDto = new HashSet<>();
+        categoryIdsForDto.add(2L);
+        createBookRequest.setCategoryIds(categoryIdsForDto);
+
+        expectedCreatedBookDto = new BookDto();
+        expectedCreatedBookDto.setTitle(createBookRequest.getTitle());
+        expectedCreatedBookDto.setAuthor(createBookRequest.getAuthor());
+        expectedCreatedBookDto.setIsbn(createBookRequest.getIsbn());
+        expectedCreatedBookDto.setCoverImage(createBookRequest.getCoverImage());
+        expectedCreatedBookDto.setDescription(createBookRequest.getDescription());
+        expectedCreatedBookDto.setCategoriesIds(categoryIdsForDto);
+        expectedCreatedBookDto.setPrice(createBookRequest.getPrice());
+    }
+
+    @SneakyThrows
+    @WithMockUser(username = "admin")
+    @DisplayName("findAll_returnsList_ok")
+    @Test
+    @Order(0)
```

Comment:
we don't need to create a particular order for our test cases. They should be isolated and give exact results without depending on the execution order. So, please, remove it and change it for all methods